### ISSUE TITLE
Reference style links are output twice

### ISSUE THREAD

**Issue by wezm** [theiskaa/mdp#2]
```markdown
In this [example link] there is a missing space after the link.

[example link]: http://www.example.com/
```

<img width="398" alt="Screenshot from 2024-10-10 09-43-02" src="https://github.com/user-attachments/assets/954f660e-188b-4de5-bf43-6324f7833b52">

**Comment by theiskaa** [comment]
You're correct! The original code on `main` uses `genpdf`, which is an outdated package and lacks support for annotations and several other features. I’ve forked the repo and started updating it to include annotations and features available in `printpdf` but missing from `genpdf`. You can find the updated repository here: [genpdf-rs-improved](https://github.com/theiskaa/genpdf-rs-improved).

In parallel, I’m integrating this updated `genpdf-rs-improved` into `mdPdf` on the `feature/styling` branch.

**Comment by theiskaa** [comment]
Thank you for your patience with this issue. I'm pleased to announce that it has been resolved with the following improvements:

- Added comprehensive styling support
- Implemented "working" annotations
- Fixed the core issue

These changes have been merged in PR #3.

I apologize for the delay in addressing this - I was in the midst of a job transition. Now that I'm settled, I'll be more responsive to issues and pull requests.

Please let me know if you encounter any problems with the implementation.

**Comment by wezm** [comment]
No worries, thanks for the fix.
